cmake/flash/CMakeLists.txt

```diff
@@ -210,6 +210,16 @@ foreach(target flash debug debugserver attach)
       ${comment}
       USES_TERMINAL
       )
+
+    # This is an artificial target to allow west to ensure all dependencies of
+    # target is build before carrying out the actual command.
+    # This allows `west` to run just the dependencies before flashing.
+    add_custom_target(west_${target}_depends
+      COMMAND ${CMAKE_COMMAND} -E echo ""
+      DEPENDS ${FLASH_DEPS}
```

Comment:
Different runner commands are not going to want FLASH_DEPS, I think. Especially `attach`, which by contract doesn't do any reflashing. Or am I missing something here?

Reply:
No you're not, but today all runners are setup in a loop:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/flash/CMakeLists.txt#L167
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/flash/CMakeLists.txt#L201-L202
```
foreach(target flash debug debugserver attach)
...
     set(cmd
      ...
      ${target}
      --skip-rebuild
      DEPENDS ${FLASH_DEPS}
              $<TARGET_PROPERTY:zephyr_property_target,FLASH_DEPENDENCIES>
```
which means all runners has exactly the same dependencies, which for some of the runners makes sense.

The main goal of the updated PR is to make things consistent between ninja and west runners.

You can argue that `attach`, `debug`, `debugserver`, and `flash` should have individual / independent dependencies (`debug` and `flash` could possibly have same dependencies).
But I found that to be outside the scope of having `west <runner>` behave identical to `ninja <runner>`.
I will be happy to look into cleaning the dependencies of the runners also.

Reply:
>  I will be happy to look into cleaning the dependencies of the runners also.

Probably in separate PR, to keep things separated.

Reply:
> which means all runners has exactly the same dependencies, which for some of the runners makes sense.

TBH I'm worried about our downstream users. If we add dependencies which do some flashing of sub-images here in this way, won't `west attach` break for nRF53 in NCS?

Reply:
By "break", I mean "do some reflashing of sub-images, which its contract says it will not do".

Reply:
that is actually a good point, but it is not `west` that breaks this contract / this PR, but the dependency in CMake.

So `ninja attach` could have side effects already today.
Cause as shown in the link, the depends is already on `master`.

Will add an extra commit to this PR, to clean the dependencies.

Reply:
Sounds good. The main thing is that we have separate dependency targets for each command, which makes sense, so we can keep the dependencies for flash an debug (which may flash the target) separate from the ones for attach (which may not).

Reply:
fixed.
Also changed the old `FLASH_DEPS` into the more correctly named `RUNNERS_DEPS` during this cleanup.